CMakeLists.txt

```diff
@@ -19,6 +19,10 @@ option(TRITON_BUILD_PYTHON_MODULE "Build Python Triton bindings" OFF)
 #  used conditionally in this file and by lit tests
 find_package(Python3 REQUIRED COMPONENTS Development Interpreter)
 
+# Customized release build type with assertions
+set(CMAKE_C_FLAGS_RELWITHDEBINFOANDASSERTIONS "-O2 -g")
+set(CMAKE_CXX_FLAGS_RELWITHDEBINFOANDASSERTIONS "-O2 -g")
```

Comment:
Could we simply merge these two lines to something like TRITON_BUILD_REL_WITH_ASSERT? considering there is a `CMAKE_BUILD_TYPE ` in CMake controlling both C and CXX.